packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -120,8 +120,27 @@ exports.sourceNodes = async (
         return
       }
       if (action === `delete`) {
-        actions.deleteNode(getNode(createNodeId(id)))
-        reporter.log(`Deleted node: ${id}`)
+        let nodesToDelete = data
+        if (!Array.isArray(data)) {
+          nodesToDelete = [data]
+        }
+
+        for (const nodeToDelete of nodesToDelete) {
+          const nodeIdToDelete = createNodeId(
+            createNodeIdWithVersion(
+              nodeToDelete.id,
+              nodeToDelete.type,
+              getOptions().languageConfig
+                ? nodeToDelete.attributes?.langcode
+                : `und`,
+              nodeToDelete.attributes?.drupal_internal__revision_id,
+              entityReferenceRevisions
+            )
+          )
+          await actions.deleteNode(getNode(nodeIdToDelete))
```

Comment:
I'm pretty sure `deleteNode` is sync — https://github.com/gatsbyjs/gatsby/blob/f4c5c482ba5da626d4f7e2d592fdc4d43eb817ba/packages/gatsby/src/redux/actions/public.js#L510

Regardless, there's no reason to wait here as Gatsby waits for any ongoing node activity before it continues.